assets/src/edit-story/components/panels/stylePanel/panel.js

```diff
@@ -0,0 +1,310 @@
+  );
+  const textOpacity = getCommonValue(selectedElements, 'textOpacity');
+
+  const {
+    state: { fonts },
+    actions: { getFontWeight, getFontFallback },
+  } = useFont();
+
+  const [state, setState] = useState({
+    backgroundColor,
+    backgroundOpacity,
+    color,
+    fontFamily,
+    fontStyle,
+    fontSize,
+    fontWeight,
+    fontFallback,
+    fontWeights,
+    textDecoration,
+    textAlign,
+    textOpacity,
+    letterSpacing,
+    lineHeight,
+    padding,
+  });
+  const [lockPaddingRatio, setLockPaddingRatio] = useState(true);
+  useEffect(() => {
+    const currentFontWeights = getFontWeight(fontFamily);
+    const currentFontFallback = getFontFallback(fontFamily);
+    setState({
+      backgroundColor,
+      backgroundOpacity,
+      color,
+      textOpacity,
+      textAlign,
+      letterSpacing,
+      lineHeight,
+      padding,
+      fontFamily,
+      fontStyle,
+      fontSize,
+      fontWeight,
+      fontWeights: currentFontWeights,
+      fontFallback: currentFontFallback,
+      textDecoration,
+    });
+  }, [
+    color,
+    textOpacity,
+    textAlign,
+    letterSpacing,
+    lineHeight,
+    padding,
+    getFontWeight,
+    fontFamily,
+    getFontFallback,
+    fontStyle,
+    fontSize,
+    fontWeight,
+    textDecoration,
+    backgroundColor,
+    backgroundOpacity,
+  ]);
+  const handleSubmit = (evt) => {
+    onSetProperties(state);
+    onSetProperties((properties) => {
+      const {
+        padding: oldPadding,
+        width,
+        height: oldHeight,
+        rotationAngle,
+        x,
+        y,
+      } = properties;
+      const { padding: newPadding } = state;
+      const updatedState = removeUnsetValues(state);
+      const newProperties = { ...properties, ...updatedState };
+      const newHeight = dataPixels(calculateTextHeight(newProperties, width));
+      const [dx, dy] = calcRotatedResizeOffset(
+        rotationAngle,
+        0,
+        0,
+        0,
+        newHeight - oldHeight
+      );
+      const ratio = getPaddingRatio(oldPadding.horizontal, oldPadding.vertical);
+      if (
+        lockPaddingRatio &&
+        (newPadding.horizontal === '' || newPadding.vertical === '') &&
+        ratio
+      ) {
+        if (newPadding.horizontal === '') {
+          newPadding.horizontal = Math.round(newPadding.vertical * ratio);
+        } else {
+          newPadding.horizontal = Math.round(newPadding.horizontal / ratio);
+        }
+      }
+      return {
+        ...updatedState,
+        height: newHeight,
+        x: dataPixels(x + dx),
+        y: dataPixels(y + dy),
+        padding: newPadding,
+      };
+    });
+    evt.preventDefault();
+  };
+
+  const getPaddingRatio = (horizontal, vertical) => {
+    if (!vertical || !horizontal) {
+      return false;
+    }
+    return horizontal / vertical;
+  };
+
+  return (
+    <SimplePanel
+      name="style"
+      title={__('Style', 'web-stories')}
+      onSubmit={handleSubmit}
+    >
+      {fonts && (
```

Comment:
Note: This file can be split more with implementing design in the design Issue, depending on how that will change.

Reply:
Moved more things out withibn 80331ec.